delivery_carrier_label_seur/model/stock.py

```diff
@@ -0,0 +1,184 @@
+
+    @api.onchange('carrier_id')
+    def carrier_id_change(self):
+        super(StockPicking, self).carrier_id_change()
+        if not self.carrier_id or self.carrier_id.carrier_type != 'seur':
+            return
+        carrier = self.carrier_id
+        self.seur_service_code = carrier.seur_service_code
+        self.seur_product_code = carrier.seur_product_code
+
+    @api.multi
+    def _generate_seur_label(self, package_ids=None):
+        self.ensure_one()
+        if not self.carrier_id.seur_config_id:
+            raise UserError(_('No SEUR Config defined in carrier'))
+        if not self.picking_type_id.warehouse_id.partner_id:
+            raise UserError(
+                _('Please define an address in the %s warehouse') % (
+                    self.warehouse_id.name))
+
+        config = self.carrier_id.seur_config_id
+
+        seur_context = {
+            'printer': 'ZEBRA',
+            'printer_model': 'LP2844-Z',
+            'ecb_code': '2C',
+        }
+
+        if config.file_type == 'pdf':
+            seur_context['pdf'] = True
+
+        data = self._get_label_data()
+        tracking_ref = False
+        label = False
+        with Picking(config.username,
+                     config.password,
+                     config.vat,
+                     config.franchise_code,
+                     'Odoo',  # seurid
+                     config.integration_code,
+                     config.accounting_code,
+                     100.0,
+                     seur_context) as picking_api:
+            tracking_ref, label, error = picking_api.create(data)
+
+            if error:
+                raise UserError(
+                    _('Error sending label to SEUR\n%s') % error)
+
+            self.carrier_tracking_ref = tracking_ref
+
+            if config.file_type == 'pdf':
+                label = label.decode('base64')
+        if tracking_ref and label:
+            return {
+                'name':
+                self.name + '_' + tracking_ref + '.' + config.file_type,
+                'file': label,
+                'file_type': config.file_type
+            }
+        else:
+            return False
+
+    def _get_label_data(self):
+        partner = self.partner_id.parent_id or self.partner_id
+        if not self.seur_service_code or not self.seur_product_code:
+            raise UserError(_(
+                'Please select SEUR service and product codes in picking'))
+        international = False
+        warehouse = self.picking_type_id and \
+            self.picking_type_id.warehouse_id or False
+
+        if self.partner_id and self.partner_id.country_id and \
+            self.partner_id.country_id.code and warehouse and \
+            warehouse.partner_id and warehouse.partner_id.country_id and \
+            warehouse.partner_id.country_id.code and \
+            self.partner_id.country_id.code != warehouse.partner_id.\
+                country_id.code:
+            international = True
```

Comment:
Maybe in a roadmap to a dedicated little module with that field on picking (condition is huge...) ?